IonDotnet/Tree/Impl/ValueFactory.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Numerics;
+
+namespace IonDotnet.Tree.Impl
+{
+    public class ValueFactory : IValueFactory
```

Comment:
I think this should be a static class with static functions.

Reply:
Unfortunately, static classes cannot implement interfaces in C#, and interfaces can't have static methods in C#.